WPLGSS/Services/SequenceRunner.cs

```diff
@@ -12,41 +12,74 @@
 namespace WPLGSS.Services
 {
     [Export(typeof(ISequenceRunner))]
-    public class SequenceRunner: ISequenceRunner
+    public class SequenceRunner : ISequenceRunner, IDisposable
     {
         public event EventHandler<StatusChangedEventArgs> SequenceRunningStateChanged;
         private System.Timers.Timer timer;
+
         ObservableCollection<Event> primarySequence;
         ObservableCollection<Event> abortSequence;
-        ConcurrentDictionary<InputChannel, double> ChannelValues;
-        public Queue<Event> cache;
+
         private WPLGSS.Models.AbortCondition AbortCondition;
-        
+        private Config config;
+        private IDataAquisition Service;
+        private Sequence sequence;
+        private TimeSpan sequenceStartTime;
+        bool abortnow = false;
 
-        // Run the abort sequence
-        public void OnThresholdReached(object source, System.Timers.ElapsedEventArgs e)
+        public SequenceRunner(IDataAquisition dataService, Config config)
         {
-            foreach(Event curr in abortSequence)
-            {
-                SequenceRunningStateChanged(this, new StatusChangedEventArgs());
-                cache.Enqueue(curr);
-            }
+            this.Service = dataService;
+            this.config = config;
         }
 
         // Raises StatusChangedEvent every millisecond and runs abort once abort condition met
         private void OnTimedEvent(object source, System.Timers.ElapsedEventArgs e)
         {
+            ObservableCollection<Channel> Channels = config.Channels;
+            ObservableCollection<Event> currSequence;
+            if (abortnow == true) currSequence = abortSequence;
+            else currSequence = primarySequence;
+
             foreach (Event curr in primarySequence)
             {
-                if (timer.Equals(curr.StartTime))
+                if ((curr.StartTime + sequenceStartTime).TotalMilliseconds - DateTime.Now.TimeOfDay.TotalMilliseconds < 1)
+                {
+                    String channelName = curr.ChannelName;
+                    Channel channel = null;
+                    foreach(Channel chan in Channels)
+                    {
+                        if (chan.Name.Equals(channelName))
+                        {
+                            channel = chan;
+                            break;
+                        }
+                            
+                    }
+                    switch(curr)
+                    {
+                        case OutputEvent output:
+                            Service.SetChannelValue(channel, 1);
+                            break;
+                        case AbortCondition abort:
+                            timer.Start();      // Restart Timer ?
```

Comment:
You don't need to re-start the timer. You just need to set `sequenceStartTime` to the current time.